spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/MaskHead.scala

```diff
@@ -0,0 +1,175 @@
+                                       resolution: Int,
+                                       scales: Array[Float],
+                                       samplingRatio: Float,
+                                       layers: Array[Int],
+                                       dilation: Int,
+                                       useGn: Boolean = false): Module[Float] = {
+
+    require(dilation == 1, s"Only support dilation = 1, but get ${dilation}")
+
+    val model = Sequential[Float]()
+    model.add(Pooler(resolution, scales, samplingRatio.toInt))
+
+    var nextFeatures = inChannels
+    var i = 0
+    while (i < layers.length) {
+      val features = layers(i)
+      // todo: not support dilation convolution
+      val module = SpatialConvolution[Float](
+        nextFeatures,
+        features,
+        kernelW = 3,
+        kernelH = 3,
+        strideW = 1,
+        strideH = 1,
+        padW = dilation,
+        padH = dilation,
+        withBias = if (useGn) false else true
+      ).setName(s"mask_fcn{${i}}")
+
+      // weight init
+      module.setInitMethod(MsraFiller(false), Zeros)
+      model.add(module)
+      nextFeatures = features
+      i += 1
+    }
+    model.add(ReLU[Float]())
+  }
+}
+
+private[nn] class MaskPostProcessor(implicit ev: TensorNumeric[Float])
+  extends AbstractModule[Table, Tensor[Float], Float] {
+
+  @transient var rangeBuffer: Tensor[Float] = null
+  private val sigmoid = Sigmoid[Float]()
+
+  /**
+    * @param input feature-maps from possibly several levels, proposal boxes and labels
+    * @return the predicted boxlists are returned with the `mask` field set
+    */
+  override def updateOutput(input: Table): Tensor[Float] = {
+    val maskLogits = input[Tensor[Float]](1)
+    val bbox = input[Tensor[Float]](2) // N * 4
+    val labels = input[Tensor[Float]](3)
+
+    val num_masks = maskLogits.size(1)
+    if (rangeBuffer == null || rangeBuffer.nElement() != num_masks) {
+      rangeBuffer = Tensor[Float](num_masks)
+      TransformerOperation.initRangeTensor(num_masks, rangeBuffer)
+    }
+
+    val mask_prob = sigmoid.forward(maskLogits)
+    require(labels.nDimension() == 1, s"Labels should be tensor with one dimention," +
+      s"but get ${labels.nDimension()}")
+    require(rangeBuffer.nElement() == labels.nElement(), s"number of masks should be same" +
+      s"with labels, but get ${rangeBuffer.nElement()} ${labels.nElement()}")
+
+    output.resize(rangeBuffer.nElement(), 1, mask_prob.size(3), mask_prob.size(4))
+
+    var i = 1
+    while (i <= rangeBuffer.nElement()) {
+      val dim = rangeBuffer.valueAt(i).toInt + 1
+      val index = labels.valueAt(i).toInt // start from 1
+      output.narrow(1, i, 1).copy(mask_prob.narrow(1, i, 1).narrow(2, index, 1))
+      i += 1
+    }
+    output
+  }
+
+  override def updateGradInput(input: Table, gradOutput: Tensor[Float]): Table = {
+    throw new UnsupportedOperationException("MaskPostProcessor only support inference")
```

Comment:
what if want to train the whole model ?

Reply:
`MaskPostProcessor` is only used for inference. It should be skipped when training.